cmd/digger/main.go

```diff
@@ -853,19 +819,39 @@ func main() {
 	log.Println("Lock provider has been created successfully")
 
 	ci := digger.DetectCI()
+
+	var logLeader = "Unknown CI"
+
 	switch ci {
 	case digger.GitHub:
+		logLeader = os.Getenv("GITHUB_ACTOR")
 		gitHubCI(lock, policyChecker, backendApi, reportStrategy)
 	case digger.GitLab:
+		logLeader = os.Getenv("CI_PROJECT_NAME")
 		gitLabCI(lock, policyChecker, backendApi, reportStrategy)
 	case digger.Azure:
+		azureContext := os.Getenv("AZURE_CONTEXT")
+		parsedAzureContext, _ := azure.GetAzureReposContext(azureContext)
```

Comment:
My only concern is that in this case the work for parsing payload will be done twice

It is a minor task but we should probably convert those parsers into singletons in the future

Should be OK to merge for now but can we make a note on top of this line with a comment which says something like "TODO: refactor this function to make it not perform double work after calling the `acureCI` function"